ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/tls/TlsConfiguration.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.besu.ethereum.api.tls;
+
+import java.nio.file.Path;
+import java.util.Objects;
+import java.util.Optional;
+
+import io.vertx.core.net.PfxOptions;
+import io.vertx.core.net.TrustOptions;
+import org.apache.tuweni.net.tls.VertxTrustOptions;
+
+public class TlsConfiguration {
+  private final Path keyStorePath;
```

Comment:
This _feels_ like a funny class - a mix of abstaction layers - i.e. base types mixed up with Vertx speciific types (PfxOptions) ... is the simplest set of things needed for RPC TLS? eg - do we need knownClientsFile AND a trustOptions?

Reply:
Yes ... sort of, I can refactor the "trustOptoins" and PfxOptions out of this class into a seperate class. It is not acting as a value object anymore i.e. keystorefile/password -> pfxOptions and knownClientsfile -> TrustOptions. I originally had a mapper class, perhaps its a good idea to bring that back.

Reply:
it has been refactored. Got rid of builder and introduced static factory methods instead.

Reply:
TlsConfiguration has been refactored.